lib/presentation/auth/widgets/enter_username.dart

```diff
@@ -115,25 +153,42 @@ class _EnterUserNameState extends State<EnterUserName> {
     );
   }
 
-  String? _validate(String? value) {
+  String? _validateFirstName(String? value) {
```

Comment:
I'm not a fan of having two functions that does the exact same except for minor differences.

Consolidate them into one function as I said earlier:
```
String? _validateName(
  String? value, {
  bool firstName = true, 
  int minLength = 2,
}) {
  if (value == null || value.trim().isEmpty) {
    return "${firstName ? 'First' : 'Last'} name is required";
  }

  if (value.length < minLength || value.length > 20) {
    return "${firstName ? 'First' : 'Last'} name should be between 4 and 60 characters long";
  }

  if (!value.startsWith(RegExp("[a-zA-Z]"))) {
    return "${firstName ? 'First' : 'Last'} name should start with a letter or number";
  }

  return null;
}
```